### PR TITLE
Merge Milestone 3 to Dev branch

### PR DESCRIPTION
**Pull request by MephistoDevelop**
Features Added:
- winning Patern
- Draw Detection
- Restart Game loop

### CODE REVIEW

File: bin/ui.rb [hunk 1]
```diff
@@ -56,9 +56,15 @@ def show_board(board)
...
-    puts "The whinner is #{board.winner}"
-    puts ''
+    puts ""
+    if(board.winner == '❌')
+      puts "The whinner is #{@@player1_name}"
+    else
+      puts "The whinner is #{@@player2_name}"
+    end
+
+    puts ""
```
Comment (suretrust) [inline bin/ui.rb]: * [x] No `puts/print` and `get.chomp` should be outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments on your code. 👇 

- [x] When there is a winner or a draw, it is not displaying the winner or informing that it's a draw, it is having the error shown below:
```ruby
bin/main.rb:19:in `<main>': undefined method `winner?' for #<Board:0x00007fa2a30d8c78> (NoMethodError)
```

* [x] All classes should be kept in `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

**Kindly submit another code review request once these issues have been fixed.** _Happy coding!_ 🎉

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress 
Your code looks nice but we need to fix a few things before we can merge.
- [ ] fix this error

- [ ] the `main.rb` should be in the `bin` folder not the` lib`

#### Submit for another review when done with the changes

### Martin Afani

**Review by fatymahmed — APPROVED** [review]
## Hi @MephistoDevelop @thcdevcl, 
### Great job making the suggested changes 🎉🎉
### It's a 👍🏾 from me!
## ✔️ Approved!
Good luck with your next projects.
 _Don't forget to submit your project completion form_

**Pull request merged**